yin_yang/plugins/konsole.py

```diff
@@ -266,3 +270,37 @@ def set_profile(service: str, profile: str):
         )
         message.setArguments([profile])
         connection.call(message)
+
+
+def set_default_profile(service: str, profile: str):
```

Comment:
This is really similar to `set_profile()`. Please merge the two into one function, f.e. `set_profile(default: bool)`

Reply:
Actually, we always call `set_profile()` and then `set_default_profile()`. So merge the two into one function, then call the function itself to change the default profile:

```python
def set_mode():
    set_profile('org.kde.konsole', profile)
    # ...

def set_profile(service: str, profile: str, default: bool = False):
    # ...
    if not default:
        set_profile(service, profile, True)
```

Reply:
Merging two functions totally make sense to me, so I have amended this PR for an extra argument in `set_profile`.
But the latter part, which suggested introducing a recursive calling, doesn't sound good to me, as it only reduces few lines of code but gains a lot more confusion for future maintenance.